pkgs/development/interpreters/python/wrapper.nix

```diff
@@ -23,17 +23,29 @@ let
       fi
       mkdir -p "$out/bin"
 
-      for path in ${stdenv.lib.concatStringsSep " " paths}; do
-        if [ -d "$path/bin" ]; then
-          cd "$path/bin"
-          for prg in *; do
-            if [ -f "$prg" ]; then
-              rm -f "$out/bin/$prg"
-              makeWrapper "$path/bin/$prg" "$out/bin/$prg" --set PYTHONHOME "$out"
-            fi
+      # Generate new wrapper scripts for the all the executables found in our
+      # dependencies.  These wrapper scripts force PYTHONHOME to point to the
+      # new package we are creating.
+      #
+      # This ends up generating wrappers for both the original executables and
+      # the pre-existing wrapper scripts.  Only the wrappers for the original
+      # executables will be used, so it doesn't hurt anything --- it's just a
+      # little distressing.
+      #
+      # The "wrapped wrappers" will not function correctly, since they might
+      # immediately set PYTHONHOME to be something else.
+      srcstores=(${stdenv.lib.concatStringsSep " " paths})
+      for srcstore in "''${srcstores[@]}"; do
+        if [[ -d "$srcstore"/bin ]]; then
+          find "$srcstore"/bin -type f -executable -print0 \
+          | while IFS= read -d "" srcfile; do
+            dstfile="$out/bin/$(basename $srcfile)"
```

Comment:
this would have been also problematic if `"$srcstore"/bin/foo/bar` and `"$srcstore"/bin/bar` exists